compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -1630,7 +1630,14 @@ impl<'a, 'tcx> euv::Delegate<'tcx> for InferBorrowKind<'a, 'tcx> {
             self.fcx.param_env,
             &place_with_id.place,
         );
+
+        let place = restrict_preicision_for_box(&place, self.capture_clause);
```

Comment:
```suggestion
        let place = restrict_precision_for_box(&place, self.capture_clause);
```

Reply:
I'm a litle bit surprised, though, to see this call *here*, in `borrow`. Doesn't this apply no matter what type of access we have, so long as we are in a move closure? I think I would expect some common helper routine invoked by `consume` *and* `borrow` to do this (along with other forms of "restrict").